ledger/participant-integration-api/src/main/scala/platform/store/FlywayMigrations.scala

```diff
@@ -153,6 +153,36 @@ private[platform] class FlywayMigrations(
       }
     }
 
+  def enforceEmptySchemaAndMigrate(
+      // TODO append-only: remove after removing support for the current (mutating) schema
+      enableAppendOnlySchema: Boolean = false
+  )(implicit resourceContext: ResourceContext): Future[Unit] =
+    dataSource.use { ds =>
+      val flyway = configurationBase(dbType, enableAppendOnlySchema, additionalMigrationPaths)
+        .dataSource(ds)
+        .ignoreFutureMigrations(false)
+        .load()
+      logger.info(
+        "Ensuring Flyway migration has either not started or there are no pending migrations..."
+      )
+      val flywayInfo = flyway.info()
+
+      flywayInfo.applied().length match {
```

Comment:
Actually this limits something that is not a problem. We have a problem if there are any offsets in the index. Should we make it so restrictive?

Reply:
I don't know if there are any other use-cases that we would be impeding other than testing migrations, though..

Reply:
I agree, but this is only used by a test ledger, so I don't really mind being a little more restrictive. The goal is to stop us doing silly things like using it to test migrations.